Doc/whatsnew/3.10.rst

```diff
@@ -171,10 +171,11 @@ These improvements are inspired by previous work in the PyPy interpreter.
 (Contributed by Pablo Galindo in :issue:`42864` and Batuhan Taskaya in
 :issue:`40176`.)
 
-:exc:`SyntaxError` exceptions raised by the intepreter will now highlight the
+:exc:`SyntaxError` exceptions now have ``end_lineno`` and
```

Comment:
One important thing to note is that the new attributes will only be populated if the information can be obtained, otherwise will be None. Given that this section is about the error messages and is possible that people interested in the extension of SyntaxError, I propose to move this to a different section (improved modules, __builtins__ maybe?).

Reply:
Wouldn't it much simpler to just add it to https://docs.python.org/3.10/whatsnew/3.10.html#other-language-changes just like other builtin changes goes?

Reply:
:+1:

Reply:
I was going to ask if these are always set, defaulting to lineno and offset.  You appear to be saying that they will always exist but might be None.  Copying lineno and offset would more standard and would be easier for IDLE and anything else that highlights by tagging a range of existing code .  In any case, SyntaxError doc should mention the defaults.

Also are other attributes always set?  Current doc suggests 'yes' but (old, from 2 decades ago?) IDLE code has backup defaults.  

I will probably close this and start over, but I want to know first about other attributes and if you will reconsider default for the new ones.

Other suggestion: new features now begins with () for with statement contexts.  Nice, but relatively minor.  Move down below Improved Error Messages, which is a good lead.

Reply:
Side comment 2: In 3.10.rst, there is a blank line between the other changes.  When I click the link above, the blank lines are gone, making the list harder to read.  I think it would likely be easier to read if the '*'s were eliminated and the items were left as  blank separated paragraphs (with maybe some other leading char that did not cause space deletion).

Reply:
> opying lineno and offset would more standard and would be easier for IDLE and anything else that highlights by tagging a range of existing code .

I prefer to set them to None, because I still need to identify if we have setted them or not and setting them to lineno and offset will technically be incorrect (getting them there may be seen as authoritative information and it will not allow to know if we have failed to locate them).

> Also are other attributes always set? 

As far as I understand yes, `offset` can be -1 when it fails to be set in some obscure places but not sure if we want to signal that publicly.

> I think it would likely be easier to read if the '*'s were eliminated and the items were left as blank separated paragraphs

Hummmm.... not sure what are you referring to exactly, unfortunately. Is the list of new syntax errors or something else?

Reply:
> As far as I understand yes, offset can be -1 when it fails to be set in some obscure places but not sure if we want to signal that publicly.

To write bullet-proof code, one needs to know all possible returns ;-).  I believe -1 makes this current code invalid.
(idlelib.runscript l.99-100)
```
            offset = getattr(value, 'offset', '') or 0
            if offset == 0:
```
Tk, with 0-based offsets, 'corrects' negatives to 0.

Reply:
I think in other What's new those are still bullet points. See for example:

https://docs.python.org/3/whatsnew/3.9.html#other-language-changes